### ISSUE TITLE
File path tab completion takes 10 seconds on mounted USB drive from Ubuntu

### ISSUE THREAD

**Issue by jbrains** [robbyrussell/oh-my-zsh#6967]
For this command:

```
$ cd /media/jbrains/USBDisk2
```

I type `cd /m`, press `<TAB>`, and it's fast and I get "media". I type `/jb`, press `<TAB>`, and it's fast and I get "jbrains". I type `/USB`, press `<TAB>` and it takes 10 seconds to get the rest of the path.

I have this problem on Ubuntu 18.04 (as Pop!_OS) over USB, but not on Mac OS 10.13.5 over Samba. The drive is connected by USB to the machine running Ubuntu and mounted using Ubuntu's automounting (not fstab). On Mac OS, I mount the same drive as `/Volumes/USBDisk2` over Samba and oh-my-zsh tab completion for file paths is immediate.

I have two USB drives connected to the machine running Ubuntu, mounted to `/media/jbrains/USBDisk1` and `/media/jbrains/USBDisk2`.

When my current working directory is inside the USB drive's file system, tab completion is immediate, but when my current working directory is not inside the USB drive's file system, then tab completion of paths on the USB drive takes 10 seconds.

```
$ uname -a
Linux stella 4.15.0-23-generic #25-Ubuntu SMP Wed May 23 18:02:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$ zsh --version
zsh 5.4.2 (x86_64-ubuntu-linux-gnu)
```

If I run `zsh` without `oh-my-zsh`, then tab completion is immediate.

If I run `oh-my-zsh` with no plugins, then tab completion still takes 10 seconds.

When tab completion is happening, when I press `Ctrl+C`, I get `Killed by signal in _path_files after 1s`, but I don't know what to try to debug `_path_files`.

What other information can I provide?

**Comment by mcornella** [comment]
You can try with a clean zsh session and sourcing OMZ's lib/ files one by one until you can reproduce the bug. You could also run `autoload -t _ls` to enable tracing of the ls completion function, which should give you a sense of where it halts.

**Comment by jbrains** [comment]
I accidentally learned something just now: the symptom came from getting stuck on a Samba share back to a MacBook Air. I noticed that `df -h` took 20 seconds, so I tried `strace df -h` and saw the process become stuck twice at reading from a Samba share to a MacBook Air on the network. When I unmounted the MacBook Air, `df -h` finished immediately and tab completion of filenames began to happen immediately.
